workflow/templateresolution/context.go

```diff
@@ -166,6 +199,17 @@ func (ctx *Context) GetTemplateScope() string {
 	return string(ctx.tmplBase.GetResourceScope()) + "/" + ctx.tmplBase.GetName()
 }
 
+// UpdateTemplateStatus update the WorkflowTemplate.Status of a given WorkflowTemplate.
+func (ctx *Context) UpdateTemplateStatus(wftmple *wfv1.WorkflowTemplate) (*wfv1.WorkflowTemplate, error) {
+	if ctx.wftmplClient == nil {
+		ctx.log.Warnln("while try update template status, wftmplClient not set")
+		return wftmple, nil
+	}
+
+	wftmple.Status.LastRunAt = metav1.Now()
+	return ctx.wftmplClient.Update(context.TODO(), wftmple, metav1.UpdateOptions{})
```

Comment:
This is going to be very hard on the kubernetes API, updating for every time we look at the template, and doing it during execution. I think this needs to go into a deduplicating queue, which is doing rate limited writes to the actual API.
One run of the [dag diamond](https://github.com/argoproj/argo-workflows/blob/main/examples/workflow-template/dag.yaml) called this 33 times.

Reply:
How about setting a ticker to run every ten minutes? The ticker can maintain a map that stores the updated workflows and their corresponding update times, ensuring updates are performed once every ten minutes.

Reply:
Thank you, I will try it later.